engine/access/rpc/engine.go

```diff
@@ -132,21 +132,26 @@ func NewBuilder(log zerolog.Logger,
 	// wrap the unsecured server with an HTTP proxy server to serve HTTP clients
 	httpServer := NewHTTPServer(unsecureGrpcServer, config.HTTPListenAddr)
 
-	// TODO: when cache size is set to 0, handle case where we do not use cache
+	var cache *lru.Cache
 	cacheSize := config.ConnectionPoolSize
-	if cacheSize == 0 {
-		cacheSize = backend.DefaultConnectionPoolSize
-	}
-	cache, err := lru.NewWithEvict(int(cacheSize), func(_, evictedValue interface{}) {
-		store := evictedValue.(*backend.CachedClient)
-		store.Close()
-		log.Debug().Str("grpc_conn_evicted", store.Address).Msg("closing grpc connection evicted from pool")
-		if accessMetrics != nil {
-			accessMetrics.ConnectionFromPoolEvicted()
+	if cacheSize > 0 {
+		// it was observed that eviction issues were occurring when the pool size was less than number of ENs + LNs
+		if cacheSize < 249 {
+			log.Debug().Msg("defaulting the connection pool size to 250")
+			cacheSize = 250
+		}
+		var err error
```

Comment:
I think the idea was to have `DefaultConnectionPoolSize` now be 250, and set the value to that